knack/config.py

```diff
@@ -47,36 +47,126 @@ def __init__(self, config_dir=None, config_env_var_prefix=None, config_file_name
         self._env_var_format = '{}{}'.format(env_var_prefix, '{section}_{option}')
         self.defaults_section_name = CLIConfig._CONFIG_DEFAULTS_SECTION
         self.config_parser.read(self.config_path)
+        self.config_file_chain = []
```

Comment:
I would recommend `self._config_file_chain`. I see this as an implementation detail and would avoid making it public.

Reply:
Updated